src/yb/util/wait_state.cc

```diff
@@ -22,6 +22,32 @@ namespace yb::util {
 // __thread WaitStateInfoPtr WaitStateInfo::threadlocal_wait_state_;
 thread_local WaitStateInfoPtr WaitStateInfo::threadlocal_wait_state_;
 
+void AUHMetadata::set_client_node_ip(const std::string &endpoint) {
+  client_node_host = 0;
+  client_node_port = 0;
+  int power = 1;
+  int index = (int)endpoint.size() - 1;
+  while (index >= 0 && endpoint[index] != ':') {
```

Comment:
prefer std existing functions to parse int for easier readability and conciseness.